src/components/add-products/add-products.js

```diff
@@ -1,20 +1,36 @@
 import React, { useState } from 'react';
 import { useDispatch } from 'react-redux';
+import { useSelector } from 'react-redux';
 
+import { productActions } from '../../store/products/action';
 import BuildForm from '../form-builder';
 import ProductCard from './product-card';
-import { productActions } from '../../store/products/action';
 
 import './product.scss';
 
 function AddProducts() {
+  const path = window.location.pathname;
+  const index = path.lastIndexOf('/');
+  const length = path.length;
+  const id = Number(path.substr(index + 1, length));
```

Comment:
you may be able to get the params through `useParams` hook
https://www.freecodecamp.org/news/a-complete-beginners-guide-to-react-router-include-router-hooks/#useparams